initfiles/bash/etc/init.d/hpcc_common.in

```diff
@@ -559,26 +559,24 @@ stop_component() {
 
     eval $stopcmd
 
+    unlock ${LOCKPATH}
+    removePid ${PIDPATH}
+
     RESULT=0
-    local counter=0
-    while [[ $RESULT -eq 0 && $counter -le 20 ]]; do
+    local waittime=30
```

Comment:
If it gives up it will throw [ Failure ] up for dali.  Then if you go to start dali again you could have init_dali running, which launched dalistop and is waiting for a return before doing kill_process.  So if you go and start then we will do a cleanup_component on the pid group that we find for COMPPIDPATH.